core/dbt/context/base.py

```diff
@@ -447,6 +447,31 @@ def log(msg: str, info: bool = False) -> str:
         else:
             logger.debug(msg)
         return ''
+    
+    @contextmember
+    @staticmethod
+    def log_var(var: str, sep = ': ', info: bool = False) -> str:
+        """Given a variable,
+        logs the variable name and value separated by sep
+
+        :param var: The variable to log
+        :param sep: how name and value are concatenated
+        :param info: If `False`, write to the log file. If `True`, write to
+            both the log file and stdout.
+
+        > macros/my_log_var_macro.sql
+
+            {% macro some_macro(arg1) %}
+              {{ log(arg1) }}
+            {% endmacro %}"
+        """
+        msg = f'{var=}' # py3.8+ only
```

Comment:
We'll obviously want to clearly document that this is only supported by python 3.8 and above. I think we may need to go a step further, though, to prevent this from bricking dbt on older python versions.

As a case in point, unit tests are running just fine for me locally (3.9) and [in the py38 GitHub Action](https://github.com/dbt-labs/dbt/pull/3912/checks?check_run_id=3634994100), but they're running into big issues in [3.6](https://github.com/dbt-labs/dbt/pull/3912/checks?check_run_id=3634994028) and [3.7](https://github.com/dbt-labs/dbt/pull/3912/checks?check_run_id=3634994062).

I wonder if something like this can do the trick?
```python
        if sys.version_info[0] == 3 and sys.verson_info[1] < 8:
            msg = f'{var}' # better than nothing? or just log "do not use this!"
        else:
            msg = f'{var=}' # py3.8+ only
```